### PR TITLE
Java: Flow summaries for Spring's Bean Properties classes

### PR DESCRIPTION
**Pull request by atorralba**
Added flow summaries for the following Spring classes in the `org.springframework.beans` package:

* [`PropertyValue`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/PropertyValue.html)
* [`PropertyValues`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/PropertyValues.html)
* [`MutablePropertyValues`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/MutablePropertyValues.html)

Tests were added following the template introduced in #6030

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    row =
+      [
+        "org.springframework.beans;PropertyValue;false;PropertyValue;(String,Object);;Argument[0];MapKey of Argument[-1];value",
+        "org.springframework.beans;PropertyValue;false;PropertyValue;(String,Object);;Argument[1];MapValue of Argument[-1];value",
+        "org.springframework.beans;PropertyValue;false;PropertyValue;(PropertyValue);;Argument[0];Argument[-1];value",
+        "org.springframework.beans;PropertyValue;false;PropertyValue;(PropertyValue,Object);;MapKey of Argument[0];MapKey of Argument[-1];value",
+        "org.springframework.beans;PropertyValue;false;PropertyValue;(PropertyValue,Object);;Argument[1];MapValue of Argument[-1];value",
+        "org.springframework.beans;PropertyValue;false;getName;;;MapKey of Argument[-1];ReturnValue;value",
+        "org.springframework.beans;PropertyValue;false;getValue;;;MapValue of Argument[-1];ReturnValue;value",
+        "org.springframework.beans;PropertyValues;true;getPropertyValue;;;MapValue of Element of Argument[-1];ReturnValue;value",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll:23]:
```suggestion
        "org.springframework.beans;PropertyValues;true;getPropertyValue;;;Element of Argument[-1];ReturnValue;value",
```

File: java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        "org.springframework.beans;PropertyValues;true;getPropertyValues;;;Element of Argument[-1];ArrayElement of ReturnValue;value",
+        "org.springframework.beans;MutablePropertyValues;true;add;(String,Object);;Argument[0];MapKey of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;add;(String,Object);;Argument[0];MapKey of Element of ReturnValue;value",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll:26]: This fact that this method returns `this` is missing. And if that is added then the library will automatically infer this line.
```suggestion
        "org.springframework.beans;MutablePropertyValues;true;add;(String,Object);;Argument[-1];ReturnValue;value",
```

File: java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        "org.springframework.beans;MutablePropertyValues;true;add;(String,Object);;Argument[1];MapValue of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;add;(String,Object);;Argument[1];MapValue of Element of ReturnValue;value",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll:28]:
```suggestion
```

File: java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValue;(PropertyValue);;Argument[0];Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValue;(PropertyValue);;Argument[0];Element of ReturnValue;value",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll:30]:
```suggestion
        "org.springframework.beans;MutablePropertyValues;true;addPropertyValue;(PropertyValue);;Argument[-1];ReturnValue;value",
```

File: java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValue;(String,Object);;Argument[0];MapKey of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValue;(String,Object);;Argument[1];MapValue of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(Map);;MapKey of Argument[0];MapKey of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(Map);;MapKey of Argument[0];MapKey of Element of ReturnValue;value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(Map);;MapValue of Argument[0];MapValue of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(Map);;MapValue of Argument[0];MapValue of Element of ReturnValue;value",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll:36]:
```suggestion
        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(Map);;MapKey of Argument[0];MapKey of Element of Argument[-1];value",
        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(Map);;MapValue of Argument[0];MapValue of Element of Argument[-1];value",
        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(Map);;Argument[-1];ReturnValue;value",
```

File: java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(PropertyValues);;MapKey of Element of Argument[0];MapKey of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(PropertyValues);;MapKey of Element of Argument[0];MapKey of Element of ReturnValue;value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(PropertyValues);;MapValue of Element of Argument[0];MapValue of Element of Argument[-1];value",
+        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(PropertyValues);;MapValue of Element of Argument[0];MapValue of Element of ReturnValue;value",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll:40]: The contained `PropertyValue` objects are copied, so we can simplify:
```suggestion
        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(PropertyValues);;Element of Argument[0];Element of Argument[-1];value",
        "org.springframework.beans;MutablePropertyValues;true;addPropertyValues;(PropertyValues);;Argument[-1];ReturnValue;value",
```

File: java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll [hunk 7]
```diff
@@ -0,0 +1,48 @@
+/**
+ * Provides classes and predicates for working with Spring classes and interfaces from
+ * `org.springframework.beans`.
+ */
+
+import java
+import semmle.code.java.dataflow.ExternalFlow
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/spring/SpringBeans.qll:7]:
```suggestion
private import semmle.code.java.dataflow.ExternalFlow
```

**Comment by atorralba** [comment]
Thank you for the review @aschackmull, all suggestions applied and tests adapted to them.

Indeed, I was overcomplicating things with some of those rows! Your optimizations work nicely, thanks 😄

**Review by aschackmull — APPROVED** [review]

**Review by aschackmull — APPROVED** [review]

**Pull request merged**